qa/rpc-tests/listtransactions.py

```diff
@@ -53,6 +62,16 @@ def run_test(self):
         check_array_result(self.nodes[1].listtransactions(),
                            {"txid":txid},
                            {"category":"receive","amount":Decimal("0.1"),"amountZat":10000000,"confirmations":1})
+        # Confirmations here are -1 instead of 0 because while we only went back
+        # 1 block, “0” means the tx is in the mempool, but the tx has already
+        # been mined and `asOfHeight` ignores the mempool regardless.
```

Comment:
Hm. Should this still be 0, to indicate "this transaction must have been in the mempool as of this height?"

Reply:
Is that true, though? Isn’t it also possible (even without `asOfHeight`) that the transaction might have been created but not yet propagated to the mempool?

And with `asOfHeight`, did we “rewind” to the moment the tx was mined (when no other tx have yet made it into the mempool) or the moment before the next tx was mined (when this tx must have been in the mempool)? To be clear, that’s a question I don’t think we should have to answer.

Also, maybe we rewound a bunch of blocks, and this tx wouldn’t have been in the mempool for five more blocks – should we be checking whether the tx happens to be in the _next_ block and return 0 in that case?

All these things make me want to stick with the simpler “ignore the mempool” behavior.